python-package/lightgbm/callback.py

```diff
@@ -71,6 +71,14 @@ class CallbackEnv:
     evaluation_result_list: Optional[_ListOfEvalResultTuples]
 
 
+def _using_cv(env: CallbackEnv) -> bool:
+    """Check if model in callback env is a CVBooster."""
+    # this string-matching is used instead of isinstance() to avoid a circular import
```

Comment:
I was thinking that it'd be expensive to do that import (since this could end up running on every iteration), but I guess it shouldn't be much overhead because Python caches imports: https://docs.python.org/3/reference/import.html#the-module-cache

I'll try that, thanks for the suggestion!

Reply:
Just pushed https://github.com/microsoft/LightGBM/pull/6761/commits/5b25aed3dbb191c473f936bdb5a78cc97e8e969e using a lazy import + `isinstance()` check.